tests/cpp/test_inlining.cpp

```diff
@@ -0,0 +1,325 @@
+
+  const auto options =
+      at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor input = at::randn({2, 3, 4}, options);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {input});
+  auto cg_outputs = fe.runFusion({input});
+
+  testValidate(&fusion, cg_outputs, {input}, __LINE__, __FILE__);
+}
+
+TEST_F(InliningTest, InliningMismatchedDims4) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeConcreteTensor({2, 3, 4});
+  fusion.addInput(tv0);
+  auto tv1 = sin(tv0);
+  auto tv2 = exp(tv1);
+  auto tv3 = relu(tv2);
+  auto tv4 = cos(tv3);
+  auto tv5 = tan(tv4);
+  fusion.addOutput(tv5);
+
+  tv3->merge(1);
+  inlineMost();
+
+  EXPECT_EQ(tv5->getComputeAtPosition(), 3);
+  EXPECT_EQ(tv4->getComputeAtPosition(), 3);
+  EXPECT_EQ(tv3->getComputeAtPosition(), 1);
+  EXPECT_EQ(tv2->getComputeAtPosition(), 1);
+  EXPECT_EQ(tv1->getComputeAtPosition(), 3);
+
+  const auto options =
+      at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor input = at::randn({2, 3, 4}, options);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {input});
+  auto cg_outputs = fe.runFusion({input});
+
+  testValidate(&fusion, cg_outputs, {input}, __LINE__, __FILE__);
+}
+
+TEST_F(InliningTest, InliningBroadcast) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeConcreteTensor({2, 3, 4});
+  fusion.addInput(tv0);
+  auto tv1 = sin(tv0);
+  // broadcasting
+  auto tv2 = broadcast(tv1, {false, true, false, true, false, true});
+  auto tv3 = cos(tv2);
+  auto tv4 = tan(tv3);
+  fusion.addOutput(tv4);
+
+  for (auto tv : {tv2, tv3, tv4}) {
+    tv->merge(0);
+    tv->merge(1);
+    tv->merge(2);
+  }
+
+  inlineMost();
+
+  EXPECT_EQ(tv4->getComputeAtPosition(), 3);
+  EXPECT_EQ(tv3->getComputeAtPosition(), 3);
+  EXPECT_EQ(tv2->getComputeAtPosition(), 3);
+  EXPECT_EQ(tv1->getComputeAtPosition(), 3);
+
+  const auto options =
+      at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor input = at::randn({2, 3, 4}, options);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {input});
+  auto cg_outputs = fe.runFusion({input});
+
+  testValidate(&fusion, cg_outputs, {input}, __LINE__, __FILE__);
+}
+
+TEST_F(InliningTest, MatchedLeafPosWithoutReplayBroadcast) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeConcreteTensor({2, 3, 4});
+  fusion.addInput(tv0);
+  auto tv1 = broadcast(tv0, {false, true, false, true, false, true});
+  auto tv2 = sin(tv1);
+  fusion.addOutput(tv2);
+
+  for (auto tv : {tv1, tv2}) {
+    tv->merge(0);
+    tv->merge(1);
+    tv->merge(2);
+  }
+
+  EXPECT_EQ(
+      TransformReplay::getMatchedLeafPosWithoutReplayPasC(tv0, tv1, 3), 3);
+  EXPECT_EQ(
+      TransformReplay::getMatchedLeafPosWithoutReplayCasP(tv1, tv0, 3), 3);
+  EXPECT_EQ(
+      TransformReplay::getMatchedLeafPosWithoutReplayPasC(tv1, tv2, 3), 3);
+  EXPECT_EQ(
+      TransformReplay::getMatchedLeafPosWithoutReplayCasP(tv2, tv1, 3), 3);
+}
```

Comment:
Moved from test_gpu3.cpp with no change except for using `EXPECT_` instead of `NVF_ERROR`.